reactor-core/build.gradle

```diff
@@ -75,12 +75,13 @@ dependencies {
 	//Optional JDK 9 Converter
 	jsr166backport "io.projectreactor:jsr166:$jsr166BackportVersion"
 
-	testCompile "junit:junit:$jUnitVersion"
-	testCompile "org.junit.jupiter:junit-jupiter-api:${jUnitJupiterVersion}"
-	testCompile "org.junit.platform:junit-platform-launcher:1.6.1"
-	testCompile "org.junit.jupiter:junit-jupiter-params:${jUnitJupiterVersion}"
-	testRuntime "org.junit.jupiter:junit-jupiter-engine:${jUnitJupiterVersion}"
-	testRuntime "org.junit.vintage:junit-vintage-engine:${jUnitJupiterVersion}"
+	implementation platform("org.junit:junit-bom:${jUnitPlatformVersion}")
```

Comment:
shouldn't it be `testImplementation` ?

Reply:
I'll try with testImplementation, but my understanding is that bom management is done in this canonical way (ie it does not add any dependencies in itself, just reads the dependencyManagement section). It's the actual adding of deps (that is still present below) that indeed uses testImplementation, testCompile, etc.

Reply:
The idea of "platforms" in Gradle is that you can manage the dependencies per scope.

By putting it in `implementation`, you're saying "Every junit-* dependency in *both* production and test code should have this version" (because the test scope inherits from the production one).

Since junit is clearly a test-only dependency, it should go to `testImplementation`.